### PR TITLE
Allow submission edits on closed forms

### PR DESCRIPTION
**Pull request by alxndrsn**
Address case 21 of https://github.com/getodk/central-backend/issues/469#issuecomment-1184672088

TODO:

* [x] confirm if `!drafts` check is required (it is)
* [x] add a test demonstrating it's necessity

### CODE REVIEW

File: lib/resources/submissions.js [hunk 1]
```diff
@@ -110,6 +110,7 @@ module.exports = (service, endpoint) => {
                     return SubmissionAttachments.upsert(extant, files);
                   }
                   // NEW SUBMISSION REQUEST: create a new submission and attachments.
+                  if(!draft && !form.acceptsSubmissions()) throw Problem.user.notAcceptingSubmissions();
```
Comment (lognaturel) [inline lib/resources/submissions.js:113]: @matthew-white we talked about linting recently. I'm surprised there isn't something to catch the lack of space after `if`?

**Comment by lognaturel** [comment]
Funny that this looked like it was going to be the hard part to me. I do think that's it!

> [Add new test cases for drafts(??)](https://github.com/getodk/central-backend/pull/530/commits/39d77e5c17263e0106bc42ee9bd5e8032c196f19)

Is there still a question around that or is the (??) stale?

I'd personally find it easier to review all of the state-related changes together once it's ready so my preference would be to squash this into one commit and cherry-pick it onto whatever branch closes the original issue. That'd be a good thing to check in with @matthew-white about.

**Comment by alxndrsn** [comment]
> Is there still a question around that or is the (??) stale?

?? is stale :slightly_smiling_face: 

Those tests also pass on `master` - it's currently an untested path.  If it's helpful, I can open a separate PR to merge them first.

**Comment by lognaturel** [comment]
Great! Makes sense to me to keep them here as you have them. As I see it, the change makes it into more of a noteworthy path. But again, pulling in @matthew-white since that touches on process for final review.

File: lib/resources/submissions.js (hunk 1 above)
Comment (alxndrsn) [reply lib/resources/submissions.js:113]: https://github.com/getodk/central-backend/issues/505 maybe?  Although there isn't currently a rule for spaces after `if`.

I'll get this fixed.

Comment (matthew-white) [reply lib/resources/submissions.js]: It does seem like it'd be helpful to add linting to CircleCI! @alxndrsn, feel free to make that change once you get the chance.

**Comment by matthew-white** [comment]
I'm realizing that a version of this issue is also present for the REST API endpoint to edit a submission (not just the OpenRosa endpoint):

https://github.com/getodk/central-backend/blob/fe1c0b78e53479285a244ec3d4b704b808af3bef/lib/resources/submissions.js#L209-L216

I'm guessing that addressing that would entail a similar change to what you've already done in this PR. What do you think about making that change to both the OpenRosa and REST APIs as part of this PR? At that point, I think it'd probably make sense to merge this PR on its own, unless @alxndrsn ends up finding it easier to put everything related to #469 in a single PR.

**Review by matthew-white — APPROVED** [review]

**Comment by matthew-white** [comment]
> I'm realizing that a version of this issue is also present for the REST API endpoint to edit a submission (not just the OpenRosa endpoint)

I'll file a separate issue about this.

**Pull request merged**